dds/DCPS/TypeAssignability.cpp

```diff
@@ -0,0 +1,2083 @@
+      case TI_PLAIN_ARRAY_LARGE:
+        return assignable_plain_array(tia, tb);
+      case TI_PLAIN_MAP_SMALL:
+      case TI_PLAIN_MAP_LARGE:
+        return assignable_plain_map(tia, tb);
+      case TI_STRONGLY_CONNECTED_COMPONENT:
+        // Does alias ever have SCC as its base type?
+        return false;
+      case EK_COMPLETE:
+        // Supporting minimal base type only
+        return false;
+      case EK_MINIMAL: {
+        const MinimalTypeObject& base_type_a = typelookup_.lookup_minimal(tia);
+        TypeObject wrapper_a(base_type_a), wrapper_b(tb);
+        return assignable(wrapper_a, wrapper_b);
+      }
+      default:
+        return false; // Future extensions
+      }
+    } else if (TK_ALIAS != ta.kind && TK_ALIAS == tb.kind) {
+      const TypeIdentifier& tib = *tb.alias_type.body.common.related_type.in();
+      switch (ta.kind) {
+      case TK_ANNOTATION:
+        return assignable_annotation(ta, tib);
+      case TK_STRUCTURE:
+        return assignable_struct(ta, tib);
+      case TK_UNION:
+        return assignable_union(ta, tib);
+      case TK_BITSET:
+        return assignable_bitset(ta, tib);
+      case TK_SEQUENCE:
+        return assignable_sequence(ta, tib);
+      case TK_ARRAY:
+        return assignable_array(ta, tib);
+      case TK_MAP:
+        return assignable_map(ta, tib);
+      case TK_ENUM:
+        return assignable_enum(ta, tib);
+      case TK_BITMASK:
+        return assignable_bitmask(ta, tib);
+      default:
+        return false; // Future extensions
+      }
+    } else if (TK_ALIAS == ta.kind && TK_ALIAS == tb.kind) {
+      const TypeIdentifier& tia = *ta.alias_type.body.common.related_type.in();
+      const TypeIdentifier& tib = *tb.alias_type.body.common.related_type.in();
+      return assignable(tia, tib);
+    }
+
+    return false;
+  }
+
+  /**
+   * @brief The first type must be TK_ANNOTATION.
+   *        The second type must not be TK_ALIAS.
+   */
+  bool TypeAssignability::assignable_annotation(const MinimalTypeObject& ta,
+                                                const MinimalTypeObject& tb) const
+  {
+    // No rule for annotation in the spec
+    return false;
+  }
+
+  /**
+   * @brief The first type must be TK_ANNOTATION.
+   *        The second type can be anything.
+   */
+  bool TypeAssignability::assignable_annotation(const MinimalTypeObject& ta,
+                                                const TypeIdentifier& tb) const
+  {
+    // No rule for annotation in the spec
+    return false;
+  }
+
+  /**
+   * @brief The first type must be TK_STRUCTURE.
+   *        The second type must not be TK_ALIAS.
+   */
+  bool TypeAssignability::assignable_struct(const MinimalTypeObject& ta,
+                                            const MinimalTypeObject& tb) const
+  {
+    if (TK_STRUCTURE != tb.kind) {
+      return false;
+    }
+
+    // Extensibility kind must match
+    TypeFlag extensibility_mask = IS_FINAL | IS_APPENDABLE | IS_MUTABLE;
```

Comment:
```suggestion
    const TypeFlag extensibility_mask = IS_FINAL | IS_APPENDABLE | IS_MUTABLE;
```
In general, make these locals variables const if it would be an error for them to change after initialization.

Reply:
Fixed this and other places.